scanners/reactnative-expo/reactnative-expo.go

```diff
@@ -0,0 +1,665 @@
+	configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.NpmStepListItem(envmanModels.EnvironmentItemModel{"command": "test"}))
+	configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.DefaultDeployStepList(false)...)
+
+	// cd
+	configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.DefaultPrepareStepList(false)...)
+	configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.NpmStepListItem(envmanModels.EnvironmentItemModel{"command": "install"}))
+
+	// eject
+	configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.NpmStepListItem(envmanModels.EnvironmentItemModel{"command": "run eject"}))
+
+	// android
+	projectLocationEnv, moduleEnv, buildVariantEnv := "$"+android.ProjectLocationInputEnvKey, "$"+android.ModuleInputEnvKey, "$"+android.BuildVariantInputEnvKey
+
+	configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.InstallMissingAndroidToolsStepListItem())
+	configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.AndroidBuildStepListItem(
+		envmanModels.EnvironmentItemModel{android.ProjectLocationInputKey: projectLocationEnv},
+		envmanModels.EnvironmentItemModel{android.ModuleInputKey: moduleEnv},
+		envmanModels.EnvironmentItemModel{android.VariantInputKey: buildVariantEnv},
+	))
+
+	// ios
+	configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.CertificateAndProfileInstallerStepListItem())
+	configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.XcodeArchiveStepListItem(
+		envmanModels.EnvironmentItemModel{ios.ProjectPathInputKey: "$" + ios.ProjectPathInputEnvKey},
+		envmanModels.EnvironmentItemModel{ios.SchemeInputKey: "$" + ios.SchemeInputEnvKey},
+		envmanModels.EnvironmentItemModel{ios.ExportMethodInputKey: "$" + ios.ExportMethodInputEnvKey},
+		envmanModels.EnvironmentItemModel{ios.ConfigurationInputKey: "Release"},
+	))
+
+	configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.DefaultDeployStepList(false)...)
+
+	bitriseDataModel, err := configBuilder.Generate(Name)
+	if err != nil {
+		return models.BitriseConfigMap{}, err
+	}
+
+	data, err := yaml.Marshal(bitriseDataModel)
+	if err != nil {
+		return models.BitriseConfigMap{}, err
+	}
+
+	configName := defaultConfigName()
+	configMap := models.BitriseConfigMap{
+		configName: string(data),
+	}
+	return configMap, nil
+}
+
+// ExcludedScannerNames ...
+func (Scanner) ExcludedScannerNames() []string {
+	return []string{
+		reactnative.Name,
+		string(ios.XcodeProjectTypeIOS),
+		string(ios.XcodeProjectTypeMacOS),
+		android.ScannerName,
+	}
+}
+
+func ensureNodeModules(cmdDir string) error {
+	log.Infof("Npm install")
+
+	cmd := command.New("npm", "install")
+	cmd.SetStdout(os.Stdout)
+	cmd.SetStderr(os.Stderr)
+	cmd.SetDir(cmdDir)
+	return cmd.Run()
+}
+
+func ejectProject(pth string, cmdDir string) error {
+	log.Infof("Eject project")
+
+	lineModification := LineModification{lastTime: time.Now(), answerCount: 0}
```

Comment:
this could be less complicated:

```
	cmd := exec.Command("bash", "./test.sh")

	stdin, err := cmd.StdinPipe()
	if err != nil {
		panic(err)
	}

	stdoutReader, err := cmd.StdoutPipe()
	if err != nil {
		panic(err)
	}

	scanner := bufio.NewScanner(stdoutReader)
	var t *time.Timer

	go func() {
		for scanner.Scan() {
			if t != nil {
				t.Stop()
			}

			line := scanner.Text()
			fmt.Println(line)

			t = time.AfterFunc(5*time.Second, func() {
				if _, err := io.WriteString(stdin, "test\n"); err != nil {
					panic(err)
				}
			})
		}
		if err := scanner.Err(); err != nil {
			panic(err)
		}
	}()

	if err := cmd.Run(); err != nil {
		panic(err)
	}
```